lighthouse-core/lib/icons.js

```diff
@@ -66,7 +66,21 @@ function pngSizedAtLeast(sizeRequirement, manifest) {
       });
 }
 
+/**
+ * @param {NonNullable<LH.Artifacts.Manifest['value']>} manifest
+ * @return {boolean} Does the manifest icons value contain at least one icon with purpose equal to "maskable"
+ */
+function containsMaskableIcon(manifest) {
+  const iconValues = manifest.icons.value;
+  return iconValues.some(icon => {
+    return icon.value.purpose &&
+      icon.value.purpose.value &&
+      icon.value.purpose.value.includes('maskable');
```

Comment:
a bit nit picky but should we check that maskable is a distinct token? [from the spec](https://www.w3.org/TR/appmanifest/#purpose-member) it looks like it's split on any whitespace and then checking each token